sql/src/planner.rs

```diff
@@ -564,16 +564,26 @@ impl<'a, P: MetaProvider> PlannerDelegate<'a, P> {
     }
 
     fn drop_table_to_plan(&self, stmt: DropTable) -> Result<Plan> {
-        let table = if stmt.if_exists {
-            stmt.table_name.to_string()
-        } else {
-            self.find_table(stmt.table_name)?.name().to_string()
-        };
+        let (table_name, partition_info) =
+            if let Some(table) = self.find_option_table(stmt.table_name.clone())? {
```

Comment:
The error returned by `find_option_table` has not been handled. Is the exception information lost here?

Reply:
Return the error with `?`.